server/models/Deployment.ts

```diff
@@ -41,7 +44,7 @@ export interface Deployment {
 
 export type DeploymentDoc = Deployment & Document<any, any, Deployment>
 
-const DeploymentSchema = new Schema<Deployment>(
+const DeploymentSchema: any = new Schema<Deployment>(
```

Comment:
try const DeploymentSchema: Schema<Deployment> = new Schema<Deployment>

Reply:
Leaving as any as Schema causes errors

> Type 'Schema<Deployment, Model<Deployment, any, any>, undefined, {}>' is not assignable to type 'Schema<Document<any, any, any>, Model<Document<any, any, any>, any, any>, undefined, {}>'.
  Types of property 'methods' are incompatible.
    Type '{ [name: string]: (this: Deployment & Document<any, any, Deployment>, ...args: any[]) => any; }' is not assignable to type '{ [name: string]: (this: Document<any, any, any>, ...args: any[]) => any; }'.
      'string' index signatures are incompatible.
        Type '(this: Deployment & Document<any, any, Deployment>, ...args: any[]) => any' is not assignable to type '(this: Document<any, any, any>, ...args: any[]) => any'.
          The 'this' types of each signature are incompatible.
            Type 'Document<any, any, any>' is not assignable to type 'Deployment & Document<any, any, Deployment>'.
              Type 'Document<any, any, any>' is missing the following properties from type 'Deployment': schemaRef, uuid, versions, metadata, and 7 more.ts(2322)

Reply:
I'd argue that typing as `any` is a backwards step. That being said, we do this already for a bunch of other schemas, so maybe worth writing up a ticket for and fixing them all at once.

Reply:
Covered in BAI-290